vim-patch:8.0.0518 and friends

justinmk opened the pull request:
different failures on travis (and appveyor) (looks like nvim is crashing) (**update:** this was a merge mistake): 

```
  ERROR   ] 1 error, listed below:
[  ERROR   ] ...jects/neovim/test/functional\legacy\045_folding_spec.lua @ 107: folding foldmethod=syntax
.\test\functional\helpers.lua:191: attempt to perform arithmetic on local 'written' (a nil value)
stack traceback:
	.\test\functional\helpers.lua:191: in function 'nvim_feed'
	.\test\functional\helpers.lua:315: in function 'feed_command'
	...jects/neovim/test/functional\legacy\045_folding_spec.lua:124: in function <...jects/neovim/test/functional\legacy\045_folding_spec.lua:107>
```

and

```
./test/functional/helpers.lua:140: ./test/functional/ui/screen.lua:440: attempt to index local 'cell' (a nil value)

stack traceback:
	./test/functional/helpers.lua:140: in function 'run'
	./test/functional/ui/screen.lua:281: in function 'wait'
	./test/functional/ui/screen.lua:216: in function 'expect'
	...eovim/neovim/test/functional/legacy/045_folding_spec.lua:95: in function <...eovim/neovim/test/functional/legacy/045_folding_spec.lua:81>

```

justinmk committed:
vim-patch:8.0.0518

Closes #7086

Problem:    Storing a zero byte from a multi-byte character causes fold text
            to show up wrong.
Solution:   Avoid putting zero in ScreenLines. (Christian Brabandt,
            closes vim/vim#1567)

https://github.com/vim/vim/commit/c6cd8409c2993b1476e123fba11cb4b8d743b896

justinmk committed:
vim-patch:8.0.0311

Problem:    Linebreak tests are old style.
Solution:   Turn the tests into new style. Share utility functions. (Ozaki
            Kiichi, closes vim/vim#1444)

https://github.com/vim/vim/commit/544d3bc9f0e494cb712a33b61558b8e8e12b1e0b

justinmk committed:
vim-patch:8.0.0090 fix breakindent bug

(original Vim commit-message is bogus)

https://github.com/vim/vim/commit/6c896867c4f5d759616028ef7cbfce2a9ed32600

justinmk committed:
vim-patch:8.0.0290 vim-patch:8.0.0394

vim-patch:8.0.0290: cursor positioning wrong if wide character wraps
Problem:    If a wide character doesn't fit at the end of the screen line, and
            the line doesn't fit on the screen, then the cursor position may
            be wrong. (anliting)
Solution:   Don't skip over wide character. (Christian Brabandt, closes vim/1408)

vim-patch:8.0.0394
Problem:    Tabs are not aligned when scrolling horizontally and a Tab doesn't
            fit. (Axel Bender)
Solution:   Handle a Tab as a not fitting character. (Christian Brabandt)
            Also fix that ":redraw" does not scroll horizontally to show the
            cursor.  And fix the test that depended on the old behavior.

https://github.com/vim/vim/commit/abc39ab642791ae3d22a524516eeedb673a95d9d

justinmk committed:
vim-patch:8.0.0126

Problem:    Display problem with 'foldcolumn' and a wide character.
            (esiegerman)
Solution:   Don't use "extra" but an allocated buffer. (Christian Brabandt,
            closes vim/vim#1310)

https://github.com/vim/vim/commit/6270660611a151c5d0f614a5f0248ccdc80ed971

justinmk committed:
vim-patch:8.0.0128

Problem:    Display test fails on MS-Windows.
Solution:   Set 'isprint' to "@".

https://github.com/vim/vim/commit/7089237885218eb8a19805bc2b75481c4efcd6ba

justinmk committed:
vim-patch:8.0.0524

Problem:    Folds are messed up when 'encodin' is "utf-8".
Solution:   Also set the fold character when it's not multi-byte.

https://github.com/vim/vim/commit/8da1e6cedf839902e15987a98733ebd31b5f1b81

justinmk commented:
QB hang:

    18:24:18,530 INFO  - # test/functional/ui/mouse_spec.lua @ 301: ui/mouse/input tab drag out of tabline to the right moves tab right

ckelsel commented on src/nvim/testdir/test_display.vim:
```diff
@@ -0,0 +1,37 @@
+" Test for displaying stuff
+if !has('gui_running') && has('unix')
+  set term=ansi
+endif
```
```TEST_FILE=test_display.res make oldtest```
**set term=asni** will be failed.

ckelsel reviewed (commented):
#7094

justinmk committed:
oldtest: cannot `:set term` in Nvim

justinmk committed:
vim-patch:8.0.0235

Problem:    Memory leak detected when running tests for diff mode.
Solution:   Free p_extra_free.

https://github.com/vim/vim/commit/b031c4ea04eb1e37a873fbb85e90d835aa1e2b1c
